File: ADT/ADTassignment_P2/Stack.java

public class Stack {
    private Student[] stu;
    private int top;

    public Stack(){
        stu = new Student[45];
        top = 0;
    }

    public void push(Student s){
        if(this.isFull()){
            return;
        }
        stu[top] = s;
        top++;
    }

    public Student pop() {
        if(this.isEmpty()){
            System.out.println("nothing left on the stack");
            return null;
        }
        Student temp = stu[top-1];
        stu[top-1] = null;
        top--;
        return temp;
    }

    public void popAll(){
        int n = this.size();
        for(int i=0;i<n;i++){
            System.out.println(this.pop());
        }
    }

    public Student peek(){
        return stu[top-1];
    }

    public boolean isEmpty(){
        return top == 0;
    }

    public boolean isFull(){
        return top >= 45;
    }

    public int size(){
        return top;
    }
}

File: ADT102/AnotherThirdAssignment/Stack.py
class Stack:
    st = []
    top = 0
    size = 0

    def __init__(self, size):
        self.size = size

    def push(self, data):
        self.st.append(data)
        self.top = self.top + 1

    def pop(self):
        temp = self.st[-1]
        del self.st[-1]
        self.top = self.top - 1
        return temp

    def peek(self):
        return self.st[-1]

    def is_empty(self):
        return self.top == 0

    def is_full(self):
        return self.top >= self.size

    def pop_all(self):
        while not self.is_empty():
            print(self.pop().__str__())

    def get_size(self):
        return self.size
